csharp/src/Apache.Arrow/DecimalUtility.cs

```diff
@@ -83,7 +83,7 @@ internal static void GetBytes(decimal value, int precision, int scale, int byteW
             int[] decimalBits = decimal.GetBits(value);
             int decScale = (decimalBits[3] >> 16) & 0x7F;
 #if NETCOREAPP
-            Span<byte> bigIntBytes = stackalloc byte[12];
+            Span<byte> bigIntBytes = stackalloc byte[13];
 
                 for (int i = 0; i < 3; i++)
                 {
```

Comment:
Another thing that should be fixed is the `stackalloc byte[4];` inside the loop.  We should just be able to stackalloc 1 time outside of the loop, and reuse the buffer space.

Reply:
fixed